File: regions/system/library/dart/codegen/lib/src/naming.dart
/// ケバブケース文字列をスネークケースに変換する。
/// 例: "auth-server" → "auth_server"
String toSnakeCase(String input) {
  return input.replaceAll('-', '_');
}

/// ケバブケース文字列をパスカルケースに変換する。
/// 例: "auth-server" → "AuthServer"
String toPascalCase(String input) {
  return input.split('-').map((segment) {
    if (segment.isEmpty) return '';
    return segment[0].toUpperCase() + segment.substring(1);
  }).join();
}

/// ケバブケース文字列をそのまま返す（正規化用）。
/// ハイフン以外の区切り文字（アンダースコア等）をハイフンに変換する。
/// 例: "auth_server" → "auth-server"
String toKebabCase(String input) {
  return input.replaceAll('_', '-').toLowerCase();
}

/// ケバブケース文字列をキャメルケースに変換する。
/// 例: "auth-server" → "authServer"
String toCamelCase(String input) {
  final pascal = toPascalCase(input);
  if (pascal.isEmpty) return '';
  return pascal[0].toLowerCase() + pascal.substring(1);
}

File: regions/system/library/typescript/codegen/src/naming.ts
/**
 * 文字列をスネークケースに変換する。
 * 例: "fooBar" → "foo_bar", "FooBar" → "foo_bar", "foo-bar" → "foo_bar"
 */
export function toSnakeCase(s: string): string {
  return s
    .replace(/([a-z0-9])([A-Z])/g, '$1_$2')
    .replace(/([A-Z]+)([A-Z][a-z])/g, '$1_$2')
    .replace(/[-\s]+/g, '_')
    .toLowerCase();
}

/**
 * 文字列をパスカルケースに変換する。
 * 例: "foo_bar" → "FooBar", "foo-bar" → "FooBar"
 */
export function toPascalCase(s: string): string {
  return s
    .replace(/[-_\s]+(.)?/g, (_, c: string | undefined) =>
      c ? c.toUpperCase() : '',
    )
    .replace(/^(.)/, (_, c: string) => c.toUpperCase());
}

/**
 * 文字列をケバブケースに変換する。
 * 例: "fooBar" → "foo-bar", "FooBar" → "foo-bar", "foo_bar" → "foo-bar"
 */
export function toKebabCase(s: string): string {
  return s
    .replace(/([a-z0-9])([A-Z])/g, '$1-$2')
    .replace(/([A-Z]+)([A-Z][a-z])/g, '$1-$2')
    .replace(/[_\s]+/g, '-')
    .toLowerCase();
}

/**
 * 文字列をキャメルケースに変換する。
 * 例: "foo_bar" → "fooBar", "foo-bar" → "fooBar", "FooBar" → "fooBar"
 */
export function toCamelCase(s: string): string {
  const pascal = toPascalCase(s);
  return pascal.charAt(0).toLowerCase() + pascal.slice(1);
}
